### PR TITLE
Cvsl 2252 afer com residential checks get com user details

### PR DESCRIPTION
**Pull request by nishanthmahasamudram**

### CODE REVIEW

File: src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsassessforearlyreleaseapi/integration/StaffResourceIntTest.kt [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+        .exchange()
+        .expectStatus()
+        .isOk
+        .expectBody(object : ParameterizedTypeReference<User>() {})
+        .returnResult().responseBody!!
+
+      Assertions.assertThat(comUserDetails).isEqualTo(comUser)
+    }
+  }
+
```
Comment (SebNegahbanMoJ) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsassessforearlyreleaseapi/integration/StaffResourceIntTest.kt:66]: **suggestion:** I think it may be worth adding a test to ensure it handles returning an empty array

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsassessforearlyreleaseapi/service/probation/DeliusApiClient.kt [hunk 2]
```diff
@@ -30,4 +30,17 @@ class DeliusApiClient(@Qualifier("oauthDeliusApiClient") val communityApiClient:
...
+    val communityApiResponse = communityApiClient
+      .post()
+      .uri("/staff")
+      .accept(MediaType.APPLICATION_JSON)
+      .bodyValue(username)
+      .retrieve()
+      .bodyToMono(typeReference<User>())
+      .block()
+    return communityApiResponse
+      ?: error("Unexpected null response from API")
```
Comment (SebNegahbanMoJ) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsassessforearlyreleaseapi/service/probation/DeliusApiClient.kt:44]: **question:** I assume this endpoint must return an empty array if there are no staff members found?

Comment (andrewrlee) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsassessforearlyreleaseapi/service/probation/DeliusApiClient.kt:44]: it returns a single user or will 404

File: src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsassessforearlyreleaseapi/integration/StaffResourceIntTest.kt (hunk 1 above)
Comment (andrewrlee) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsassessforearlyreleaseapi/integration/StaffResourceIntTest.kt:66]: worth static importing Assertions, but a test to cover the 404 would be good.

Comment (nishanthmahasamudram) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsassessforearlyreleaseapi/integration/StaffResourceIntTest.kt:87]: done.

**Review by andrewrlee — APPROVED** [review]
LGTM
